### DISCUSSION TITLE
Replace Swift Mailer in Craft 4

### DISCUSSION THREAD

**Discussion by bencroker** [craftcms/cms#10062] [Ideas] [answered]
[Swift Mailer](https://swiftmailer.symfony.com/docs/introduction.html), besides being a PITA to work with, is no longer maintained as of November 2021. It has been replaced by [Symfony Mailer](https://symfony.com/doc/current/mailer.html), which is a more modern take on a mailer component, by the same author. 

Fortunately, _"Migrating from Swiftmailer to Symfony Mailer is a relatively easy task as the concepts are the same between the two projects"_, so perhaps Craft 4 would be a good time to make the switch.

From the blog article at https://symfony.com/blog/the-end-of-swiftmailer:

> For many years, I've tried to work on modernizing the Swiftmailer code base one step at a time by trying to introduce namespaces, move to the Composer autoloader, trying to remove the special (and heavy) "boot sequence", trying to switch to the Symfony dependency injection and event dispatcher components. But all my efforts always finished as dead ends, probably because the task was too important for little benefits as Swiftmailer suffers from many more conceptual issues: it has a weird and complex class inheritance and message instances are not data objects which makes it painful to serialize them. It also lacks some modern features like supporting Twig for composing emails and support for third-party providers. Both features could have been added to the current code base, but it would not have been ideal. Another issue is the way Swiftmailer uses a "spool" mechanism to support asynchronous emails; it barely works and it's not very flexible. I can go on and on, but you get the point.
>
> So, at some point, I decided to start again, but with a very different and radical approach. Instead of trying to modernize the code one step at a time, I've copied the code and started to first work on isolating a Symfony Mime component. Then, I refactored the rest of the code base as much as possible to create Symfony Mailer. The result is Symfony Mailer. Symfony Mailer has quite a lot of code in common with Swiftmailer but none of its idiosyncrasies.

Symfony Mailer also supports third-party transports that will make updating plugins that provide transport adapters straightforward.

<img width="527" alt="Screenshot 2021-11-06 at 19 30 09" src="https://user-images.githubusercontent.com/57572400/140622980-6f89a9fa-3af8-4c83-afbf-820a26fdd928.png">

**Comment by engram-design** [comment]
Here's my +1 for this! SwiftMailer has always felt pretty archaic to me (why can't I bloody use `$message->getHtmlBody()`), and getting useful errors from it https://github.com/craftcms/cms/pull/8091

**Comment by brandonkelly** [comment, marked as answer]
Symfony Mailer is now included in Craft 4 🎉  (#10568)

**Reply by engram-design** [reply]
Amazing mate! Love it.

**Reply by leevigraham** [reply]
Nice.
